### PR TITLE
Update to RxJava 3 + AutoDispose 2

### PR DESCRIPTION
**Pull request by ShaishavGandhi**
Ref #56

### CODE REVIEW

File: rxdogtag-autodispose/gradle.properties [hunk 1]
```diff
@@ -17,4 +17,4 @@
 POM_NAME=RxDogTag (AutoDispose Support)
 POM_ARTIFACT_ID=rxdogtag-autodispose
 POM_PACKAGING=jar
-AUTOMATIC_MODULE_NAME=com.uber.rxdogtag.autodispose
+AUTOMATIC_MODULE_NAME=com.uber.rxdogtag2.autodispose2
```
Comment (ShaishavGandhi) [inline rxdogtag-autodispose/gradle.properties]: Not sure how module name works so wasn't sure if this is correct. Let me know

File: rxdogtag/gradle.properties [hunk 2]
```diff
@@ -17,4 +17,4 @@
 POM_NAME=RxDogTag
 POM_ARTIFACT_ID=rxdogtag
 POM_PACKAGING=jar
-AUTOMATIC_MODULE_NAME=com.uber.rxdogtag
+AUTOMATIC_MODULE_NAME=com.uber.rxdogtag2
```
Comment (ShaishavGandhi) [inline rxdogtag/gradle.properties]: Same here. Let me know if this tracks package name or coords

File: rxdogtag/src/test/java/anotherpackage/ObserverHandlerDefaultsTest.java [hunk 3]
```diff
@@ -60,8 +60,8 @@ public void defaultsShouldReturnSameObserver() {
    */
   @Test
   public void verifyPackages() {
-    checkPackage("Observable", "io.reactivex", Observable.class.getPackage().getName());
-    checkPackage("RxDogTag", "com.uber.rxdogtag", RxDogTag.class.getPackage().getName());
+    checkPackage("Observable", "io.reactivex.rxjava3", "io.reactivex.rxjava3");
```
Comment (ShaishavGandhi) [inline rxdogtag/src/test/java/anotherpackage/ObserverHandlerDefaultsTest.java]: This test is probably useless now

File: rxdogtag/src/main/java/rxdogtag2/RxDogTag.java [hunk 4]
```diff
@@ -518,9 +517,10 @@ public void install() {
     /** Default ignored packages when sourcing originating subscribe points. */
     private static final Collection<String> DEFAULT_IGNORED_PACKAGES =
         asList(
-            // "io.reactivex"
-            Observable.class.getPackage().getName(),
-            // "com.uber.rxdogtag"
+            // RxJava 3 package name. Referencing by type would have been nice but
+            // io.reactivex.rxjava3 doesn't have any base classes.
+            "io.reactivex.rxjava3",
```
Comment (ShaishavGandhi) [inline rxdogtag/src/main/java/rxdogtag2/RxDogTag.java]: Observable.class.getPackage would now return `io.reactivex.rxjava3.core`. There are no classes in `io.reactivex.rxjava3` so we would have to refer to this loosely typed String

**Comment by ShaishavGandhi** [comment]
> Let's go with just "rxdogtag2" for the package name like we did with autodispose. Artifact names can stay the same, only the group ID needs to change

That is what I did. `rxdogtag2` is the new package name and groupId is now `com.uber.rxdogtag2`

File: rxdogtag/gradle.properties (hunk 2 above)
Comment (ZacSweers) [reply rxdogtag/gradle.properties]: I think this should just be rxdogtag2 like the package name

Comment (ShaishavGandhi) [reply rxdogtag/gradle.properties]: Fixed 1866678

File: rxdogtag-autodispose/gradle.properties (hunk 1 above)
Comment (ShaishavGandhi) [reply rxdogtag-autodispose/gradle.properties]: 1866678

**Comment by ZacSweers** [comment]
I approve, the GitHub mobile app is just being bad. Merge at will

**Pull request merged**